Apps/WebClient/src/Server/Controllers/WalletStatusController.cs

```diff
@@ -0,0 +1,128 @@
+    using HealthGateway.WebClient.Services;
+    using Microsoft.AspNetCore.Authorization;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.Extensions.Logging;
+
+    /// <summary>
+    /// Provides a web hook for Aca-Py to receive status updates on Wallet Connections and Credentials.
+    /// </summary>
+    [ApiVersion("1.0")]
+    [Route("v{version:apiVersion}/api/[controller]")]
+    [ApiController]
+    [IgnoreAudit]
+    [TypeFilter(typeof(AvailabilityFilter))]
+    public class WalletStatusController : Controller
+    {
+        private readonly ILogger logger;
+        private readonly IWalletStatusService walletStatusService;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WalletStatusController"/> class.
+        /// </summary>
+        /// <param name="logger">Injected Logger Provider.</param>
+        /// <param name="walletStatusService">The injected wallet status service provider.</param>
+        public WalletStatusController(ILogger<WalletStatusController> logger, IWalletStatusService walletStatusService)
+        {
+            this.logger = logger;
+            this.walletStatusService = walletStatusService;
+        }
+
+        /// <summary>
+        /// Handle webhook events sent from the issuing agent.
+        /// </summary>
+        /// <param name="topic">The type of webhook response (connection or issue credential).</param>
+        /// <param name="data">Webhook response data.</param>
+        /// <returns>An empty response.</returns>
+        /// <response code="204">Webhook request received.</response>
+        [HttpPost]
+        [Route("topic/{topic}")]
+        [Authorize(Policy = ApiKeyPolicy.Write)]
+        public IActionResult Webhook(string topic, [FromBody] WebhookData data)
+        {
+            this.logger.LogInformation("Webhook topic \"{topic}\"", topic);
+            switch (topic)
+            {
+                case WebhookTopic.Connections:
+                    this.HandleConnectionUpdateAsync(data);
+                    break;
+                case WebhookTopic.IssueCredential:
+                    this.HandleIssueCredentialUpdateAsync(data);
+                    break;
+                case WebhookTopic.RevocationRegistry:
+                    break;
+                case WebhookTopic.BasicMessage:
+                    this.logger.LogInformation("Basic Message data: for {@JObject}", JsonSerializer.Serialize(data));
+                    break;
+                default:
+                    this.logger.LogInformation("Webhook {topic} is not supported", topic);
+                    break;
+            }
+
+            return this.NoContent();
+        }
+
+        // Handle webhook events for connection states.
+        private void HandleConnectionUpdateAsync(WebhookData data)
+        {
+            this.logger.LogInformation("Connection state \"{state}\" for {@JObject}", data.State, JsonSerializer.Serialize(data));
```

Comment:
I think we likely use Debug or Trace here since this could potentially have PI ?!

Reply:
+1

Reply:
Cleaned up all the logging to be appriopriate as it was likely going to be too chatty.